activity/journal.go

```diff
@@ -0,0 +1,119 @@
+package activity
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Journal is used to describe a log of activity. That activity is associated with a user-generated
+// event by the ID, and can include a description and start time to provide further context when
+// listed in activity history.
+type Journal struct {
+	ID, Description string
+	start           time.Time
+}
+
+// NewJournal creates a new journal entry with start time set to 'now'.
+func NewJournal(id, desc string) Journal {
+	// We set Start based on the time we first encounter the process, not when the process was
+	// started. This makes history make a little more sense when interacting with things like
+	// the shell, which was likely started before wash was.
+	return Journal{ID: id, Description: desc, start: time.Now()}
+}
+
+type historyBlob struct {
```

Comment:
I like what you did with the `HistoryResponse` type in the API -- it's easy to think of history as an array of activities. It would be nice if `historyBlob` could reflect that symmetry, but it's not a blocker/could be something that's done later.

Reply:
I'm not sure what that would look like. You mean in terms of naming?

Reply:
That and the type definition. So something like
```
type history struct {
        mux           sync.Mutex
	activities   []Activity
	stored       map[string]int
}
```

where activity may look like:
```
type Activity struct {
       start     time.Time
       journal Journal
}
```